lib/Doctrine/DBAL/Types/DateTimeType.php

```diff
@@ -49,6 +49,10 @@ public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $pla
      */
     public function convertToDatabaseValue($value, AbstractPlatform $platform)
     {
+        if ($value !== null && !$value instanceof \DateTime) {
```

Comment:
This is actually correct: `null` is an allowed value. Maybe applying De Morgan would make it more readable tho